src/Symfony/Security/State/AccessCheckerProvider.php

```diff
@@ -75,10 +73,31 @@ public function provide(Operation $operation, array $uriVariables = [], array $c
             ];
         }
 
-        if (!$this->resourceAccessChecker->isGranted($operation->getClass(), $isGranted, $resourceAccessCheckerContext)) {
+        if ($isGranted && !$this->resourceAccessChecker->isGranted($operation->getClass(), $isGranted, $resourceAccessCheckerContext)) {
             $operation instanceof GraphQlOperation ? throw new AccessDeniedHttpException($message ?? 'Access Denied.') : throw new AccessDeniedException($message ?? 'Access Denied.');
         }
 
+        if ($operation instanceof HttpOperation && $operation->getUriVariables()) {
```

Comment:
same thing what about having this feature optional at first, therefore we could put this logic in a provider decorating the `AccessCheckerProvide`. As both the ReadProvider and this part are quite tighten to security I'd leave them into this namespace.